src/controllers/BotController.ts

```diff
@@ -0,0 +1,134 @@
+  botService: BotService;
+  constructor() {
+    this.botService = new BotService();
+  }
+
+  @Get("/:botId/environments")
+  getEnvironments(@Param("botId") botId: string): GetEnvironmentsResponseBody {
+    const environments = this.botService.findEnvironments(botId);
+    if (environments && environments.length < 1) {
+      throw new NotFoundError(`No environments found for the bot.`);
+    }
+    return environments;
+  }
+
+  @Get("/:botId/environments/:environmentId/state")
+  getState(
+    @Param("botId") botId: string,
+    @Param("environmentId") environmentId: Environment
+  ): GetStateResponseBody {
+    const botState = this.botService.getState(botId, environmentId);
+    if (!botState) {
+      throw new NotFoundError(`Bot has no state for this environment.`);
+    }
+    return { state: botState, version: "1.0" };
+  }
+
+  @Put("/:botId/environments/:environmentId/conversations/:convId")
+  @OnUndefined(204)
+  putConversation(
+    @Param("botId") botId: string,
+    @Param("environmentId") environmentId: Environment,
+    @Param("convId") convId: string,
+    @Body() body: PutConversationRequestBody
+  ): void {
+    // check if the conversation exists in cache else create it and save information
+    const creationSuccessful = this.botService.createConversation(
+      botId,
+      environmentId as Environment,
+      convId,
+      body.context,
+      body.sdes || undefined // SDES can be found only if the bot was enabled to fetch engagement in TPB
+    );
+
+    if (!creationSuccessful) {
+      throw new InternalServerError(`Conversation creation was not success`);
+    }
+  }
+
+  @Post("/:botId/environments/:environmentId/conversations/:convId/events")
+  @OnUndefined(404)
+  fetchConversationEvents(
+    @Param("botId") botId: string,
+    @Param("environmentId") environmentId: Environment,
+    @Param("convId") convId: string,
+    @Body() body: PostConversationEventRequestBody
+  ): PostConversationEventsResponseBody {
+    const bot = this.botService.findBot(botId, environmentId);
+    // Bot was not found
+    if (!bot) {
+      throw new NotFoundError(`Bot not found for the given environment.`);
+    }
+
+    const conversation = this.botService.getConversation(
+      botId,
+      environmentId,
+      convId
+    );
+
+    if (!conversation) {
+      throw new NotFoundError(`no conversation found for the given bot.`);
+    }
+
+    const { type, data } = body;
+    // Currently we only support Text and Rich Content events sent by customer
+
+    switch (type) {
+      case EventMessageType.RICH_CONTENT:
+        return this.botService.fetchConversationRichContentEvents(
+          botId,
+          data as IRichContentEventData
+        ) as PostConversationEventsResponseBody;
+        break;
+      case EventMessageType.TEXT:
+        return this.botService.fetchConversationTextEvents(
+          botId,
+          data as ITextEventData
+        );
+        break;
+      case EventMessageType.START:
```

Comment:
Welcome Event is also cloaked as START event

Reply:
What you mean i don't understand?